### PR TITLE
1643128: Do not execute subscription-manager dnf plugin twice; ENT-987

### PR DESCRIPTION
**Pull request by jirihnidek**
* BZ: https://bugzilla.redhat.com/show_bug.cgi?id=1643128
* `config()` is name of hook.

### CODE REVIEW

**Comment by jirihnidek** [comment]
@wottop this bug was introduced in this commit: https://github.com/candlepin/subscription-manager/commit/4cee072dddd2df09b1d4076e978cba8cea2b2154. Please do PR review.

File: src/dnf-plugins/subscription-manager.py [hunk 1]
```diff
@@ -54,9 +54,9 @@ def __init__(self, base, cli):
         super(SubscriptionManager, self).__init__(base, cli)
         self.base = base
         self.cli = cli
-        self.config()
+        self._config()
 
-    def config(self):
+    def _config(self):
```
Comment (cnsnyder) [inline src/dnf-plugins/subscription-manager.py]: Won't this cause the plugin not to run on any dnf hook?

File: src/dnf-plugins/subscription-manager.py [hunk 2]
```diff
@@ -54,9 +54,9 @@ def __init__(self, base, cli):
         super(SubscriptionManager, self).__init__(base, cli)
         self.base = base
         self.cli = cli
-        self.config()
+        self._config()
```
Comment (cnsnyder) [inline src/dnf-plugins/subscription-manager.py]: Maybe we should just not call this method on __init__.

File: src/dnf-plugins/subscription-manager.py (hunk 1 above)
Comment (jirihnidek) [reply src/dnf-plugins/subscription-manager.py]: The `_config()` will be triggered. The `__init__()` is sort of hook in this case. This 'hook' is triggered as the first.

File: src/dnf-plugins/subscription-manager.py (hunk 2 above)
Comment (jirihnidek) [reply src/dnf-plugins/subscription-manager.py]: I already mentioned this in this comment: https://github.com/candlepin/subscription-manager/pull/2000#issuecomment-448604852

See also this: https://github.com/candlepin/subscription-manager/pull/1884 and this: https://bugzilla.redhat.com/show_bug.cgi?id=1609052

**Pull request merged**